csrc/fusion.cpp

```diff
@@ -242,6 +242,11 @@ void Fusion::addInput(Val* input) {
         "Immediate scalar value cannot be added as an input. It is not necessary to pass it as an input.");
   }
 
+  NVF_CHECK(
+      std::find(inputs_.begin(), inputs_.end(), input) == inputs_.end(),
```

Comment:
I think this is equivalent and shorter. 

```suggestion
      input->isFusionInput(),
```

Reply:
I was debating on this earlier. `isFusionInput` only checks on the `is_fusion_input_` flag and I figured that we are really wanting to avoid duplicate in the `inputs_` vector.

Admittedly it's supposed to be equivalent. :man_shrugging: I'll switch for cleaner code.